tests/main/views/test_direct_award.py

```diff
@@ -1228,3 +1240,305 @@ def test_direct_award_outcome_scenarios(
+            data=json.dumps(self._updated_by_data()),
+            content_type="application/json",
+        )
+        assert res.status_code == 404
+        assert json.loads(res.get_data()) == {
+            "error": "Project 952738338097614 not found",
+        }
+
+    # tuples of paramaters in the order:
+    #     has_active_search
+    #     has_non_active_search
+    #     project_locked
+    #     existing_outcome_result
+    #     existing_outcome_complete
+    #     expected_status_code
+    #     expected_response_data
+    _base_direct_award_nonawarded_scenarios = (
+        # "happy" paths
+        (True, True, True, None, False, 200, _anydict,),
+        (True, False, True, None, False, 200, _anydict,),
+        (False, True, True, "cancelled", False, 200, _anydict,),
+        (True, False, True, "awarded", False, 200, _anydict,),
+        (True, True, True, "cancelled", False, 200, _anydict,),
+        (True, False, False, "none-suitable", False, 200, _anydict),
+        (False, False, False, None, False, 200, _anydict),
+        # "failure" paths
+        (
+            True,
+            True,
+            True,
+            "cancelled",
+            True,
+            410,
+            {
+                "error": AnyStringMatching(r"Project \d+ already has a completed outcome: \d+"),
+            },
+        ),
+        (
+            True,
+            False,
+            True,
+            "awarded",
+            True,
+            410,
+            {
+                "error": AnyStringMatching(r"Project \d+ already has a completed outcome: \d+"),
+            },
+        ),
+    )
+
+    @pytest.mark.parametrize(
+        (
+            "endpoint_rel_path",
+            "has_active_search",
+            "has_non_active_search",
+            "project_locked",
+            "existing_outcome_result",
+            "existing_outcome_complete",
+            "expected_status_code",
+            "expected_response_data",
+        ),
+        tuple(chain(
+            (
+                # we test all of the _base_direct_award_nonawarded_scenarios against the known valid nonawarded reasons
+                (endpoint_rel_path,) + base_scenario
+                for endpoint_rel_path, base_scenario in product(
+                    ("none-suitable", "cancel",),
+                    _base_direct_award_nonawarded_scenarios,
+                )
+            ),
+            (
+                # but we also include a couple of specific cases with invalid reasons
+                (
+                    "severe-chill",
+                    True,
+                    False,
+                    True,
+                    None,
+                    False,
+                    404,
+                    {
+                        "error": AnyStringMatching(r".*the requested url was not found.*", re.I),
+                    },
+                ),
+                (
+                    "awarded",
```

Comment:
Okay this kinda makes sense now but probably needs a comment - "Check that `awarded` is not a URI because awarding is done against a project URI (or, in future a brief)" or sommat like that maybe.